ql/src/java/org/apache/hadoop/hive/ql/parse/TransformSpec.java

```diff
@@ -61,4 +75,18 @@ public Optional<Integer> getTransformParam() {
   public void setTransformParam(Optional<Integer> transformParam) {
     this.transformParam = transformParam;
   }
+
+  public static TransformType fromString(String transformString) {
```

Comment:
```` 
public enum TransformType {
    IDENTITY, 
    ....
    BUCKET, 
    VOID
  }
 
TransformType fromString(String value) {
    return TransformSpec.valueOf(value.toUppercase()).getTransformType()
}
````

Reply:
I wish this was that simple. `transform.toString()` doesn't guarantee that it provides the keyword of the transform only, it provides much more than that.
A `toString()` called on a bucket transform would return something like this - 
https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/transforms/Bucket.java#L138-L140

Reply:
Implemented something similar to Iceberg's implementation here - 
https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/transforms/Transforms.java#L40-L69

Done.